sdks/python/src/opik/cli/imports/experiment.py

```diff
@@ -0,0 +1,541 @@
+
+
+def load_experiment_data(experiment_file: Path) -> Dict[str, Any]:
+    """Load experiment data from JSON file."""
+    with open(experiment_file, "r", encoding="utf-8") as f:
+        return json.load(f)
+
+
+def recreate_experiment(
+    client: opik.Opik,
+    experiment_data: Dict[str, Any],
+    project_name: str,
+    trace_id_map: Optional[Dict[str, str]] = None,
+    dry_run: bool = False,
+    debug: bool = False,
+) -> bool:
+    """Recreate a single experiment from exported data.
+
+    Note: This function expects that traces have already been imported into the target workspace.
+    When traces are imported, they receive new IDs. The trace IDs from the exported experiment
+    data may not match the imported traces unless they were imported in the same session and
+    the trace IDs were preserved during import. Items referencing non-existent traces will be skipped.
+    """
+    experiment_info = experiment_data["experiment"]
+    items_data = experiment_data["items"]
+
+    experiment_name = (
+        experiment_info.get("name") or f"recreated-{experiment_info['id']}"
+    )
+    dataset_name = experiment_info["dataset_name"]
+
+    console.print(f"[blue]Recreating experiment: {experiment_name}[/blue]")
+
+    if dry_run:
+        console.print(
+            f"[yellow]Would create experiment '{experiment_name}' with {len(items_data)} items[/yellow]"
+        )
+        return True
+
+    try:
+        # Get or create the dataset
+        dataset = client.get_or_create_dataset(
+            name=dataset_name,
+            description=f"Recreated dataset for experiment {experiment_name}",
+        )
+
+        # Create the experiment
+        experiment = client.create_experiment(
+            dataset_name=dataset_name,
+            name=experiment_name,
+            experiment_config=experiment_info.get("metadata"),
+            type=experiment_info.get("type", "regular"),
+        )
+
+        # Process experiment items
+        experiment_items = []
+        successful_items = 0
+        skipped_items = 0
+
+        for item_data in items_data:
+            # Handle trace reference (from deduplicated exports)
+            trace_id = handle_trace_reference(item_data)
+            if not trace_id:
+                console.print(
+                    "[yellow]Warning: No trace ID found, skipping item[/yellow]"
+                )
+                skipped_items += 1
+                continue
+
+            # Translate trace id from source (workspace A) to newly created trace id (workspace B)
+            new_trace_id = translate_trace_id(trace_id, trace_id_map)
+            if not new_trace_id:
+                console.print(
+                    f"[yellow]Warning: No mapping for trace {trace_id}. Skipping item.[/yellow]"
+                )
+                skipped_items += 1
+                continue
+
+            # Handle dataset item
+            dataset_item_data = item_data.get("dataset_item_data")
```

Comment:
Minor: wouldn't it be easier and more straightforward to perform full dataset exports and imports? Instead of recreating the dataset from its pieces in experiment items? IDs mapping could be handled in the same way as for the traces.

It's only my recommendation, up to you how to keep it, but I think we're not that far from just reusing a piece of the functionality which we already have in the same PR.